janus_core/cli/geomopt.py

```diff
@@ -157,45 +235,23 @@ def geomopt(
     else:
         write_kwargs["filename"] = f"{s_point.struct_name}-opt.xyz"
 
-    if "opt_kwargs" in minimize_kwargs:
-        # Check trajectory path not duplicated
-        if "trajectory" in minimize_kwargs["opt_kwargs"]:
-            raise ValueError("'trajectory' must be passed through the --traj option")
-    else:
-        minimize_kwargs["opt_kwargs"] = {}
+    _set_minimize_kwargs(minimize_kwargs, traj, vectors_only, pressure)
 
-    if "traj_kwargs" not in minimize_kwargs:
-        minimize_kwargs["traj_kwargs"] = {}
-
-    # Set same trajectory filenames to overwrite saved binary with xyz
-    if traj:
-        minimize_kwargs["opt_kwargs"]["trajectory"] = traj
-        minimize_kwargs["traj_kwargs"]["filename"] = traj
-
-    # Check hydrostatic_strain and scalar pressure not duplicated
-    if "filter_kwargs" in minimize_kwargs:
-        if "hydrostatic_strain" in minimize_kwargs["filter_kwargs"]:
-            raise ValueError(
-                "'hydrostatic_strain' must be passed through the --vectors-only option"
-            )
-        if "scalar_pressure" in minimize_kwargs["filter_kwargs"]:
+    if fully_opt or vectors_only:
+        # Use default filter unless filter function explicitly passed
+        fully_opt_dict = {"filter_func": filter_func} if filter_func else {}
+    else:
+        if filter_func:
```

Comment:
```suggestion
    elif filter_func:
```

Reply:
Ah, it's hard to see in GitHub that that line is inside the block (one reason I hate Python's white-space delimiting)